__tests__/roundtrip_test.ml

```diff
@@ -13,6 +11,40 @@ let run_test ~name ~write ~read ~data =
     |> toEqual data'
   )
 
+let run_read_test ~name ~read ~data ~expected =
+  let open Expect in
+  let decode = Atdgen_codec_runtime.Decode.decode read in
+  let data' = decode data in
+  test name (fun () ->
+    expect data'
+    |> toEqual expected
+  )
+
+
+(** Test that decoding data that wouldn't normally be encoded still equals the correct value *)
+let () =
+    describe "read tests" (fun () ->
+      run_read_test
+        ~name:"nullable field set to null"
+        ~read:Test_bs.read_optional_field
+        ~expected:{with_default = 9; no_default = None; no_default_nullable = None}
+        ~data:begin
+          let input = Js.Dict.empty () in
+          let () = Js.Dict.set input "no_default_nullable" Js.Json.null in
+          Js.Json.object_ input
+        end;
```

Comment:
Could use:

```suggestion
        ~data:[%raw {|{no_default_nullable:null}|}]
```

Reply:
There are many ways to do this, I figured I had to use a type safe way 😅 I'll change both